drivers/overlay/overlay.go

```diff
@@ -909,13 +914,18 @@ func (d *Driver) Diff(id string, idMappings *idtools.IDMappings, parent string,
 		return nil, err
 	}
 
+	whiteoutFormat := archive.OverlayWhiteoutFormat
+	if d.options.mountProgram != "" {
```

Comment:
Could you add a comment here? It seems like there's some implicit condition when `mountProgram != ""`.

Reply:
I've pushed a new version with a comment (and the common code block refactored in a function)

Reply:
Thanks a lot, @giuseppe !